queries/haskell/highlights.scm

```diff
@@ -190,26 +176,44 @@
          ]))
      (quasiquote)
      ((exp_name) . (operator))
-    ]))
+    ])+
+ (#eq? @variable @_name))
+
+((signature name: (variable) @variable)
+ . (comment)*
+ . (function 
+    name: (variable) @_name
+    rhs: (exp_infix 
+      [
+       (exp_literal)
+       (exp_apply 
+         (exp_name 
+           [(constructor)
+            (variable)
+           ]))
+       (quasiquote)
+       ((exp_name) . (operator))
+      ]))+
```

Comment:
This section looks to me like a nearly exhaustive list of available function rhs.  I think we can compromise and just check for 
```
(function
  name: (variable)
  rhs: (exp_infix))
```
for this section of the query, instead of going deep down.

Reply:
I've tried that. IIRC, it matched too much.
I'd have to play around with that again after work.

Reply:
Shall we create a separate issue from this thread?
I think preventing the segfault is quite high priority.

Reply:
Yeah preventing the segfault is higher priority for sure.